darkmoon/api/v1/metadata/views.py

```diff
@@ -0,0 +1,205 @@
+"""This is the router file."""
+
+###########
+# IMPORTS #
+###########
+
+from typing import Optional
+
+import bson
+from bson.objectid import ObjectId
+from fastapi import APIRouter, Depends, HTTPException
+from motor.motor_asyncio import AsyncIOMotorCollection
+from pymongo import errors
+
+from darkmoon.api.v1.metadata.schema import Metadata, MetadataEntity
+from darkmoon.core.database import get_file_metadata_collection
+
+####################
+# GLOBAL VARIABLES #
+####################
+
+router = APIRouter(prefix="/metadata", tags=["metadata"])
+
+###########
+# CLASSES #
+
+
+#############
+# FUNCTIONS #
+#############
+
+
+@router.get("/")
+async def list_metadata(
+    collection: AsyncIOMotorCollection = Depends(get_file_metadata_collection),
+    file_name: Optional[str] = None,
+    hash_type: Optional[str] = None,
+    hash: Optional[str] = None,
+) -> list[MetadataEntity]:
+    """Return list of metadata that matches the parameters in the database.
+
+    Parameters:
+        file_name (Optional[str]): The name of the file being
+            searched. Is None by default.
+        hash_type (Optional[str]): The type of hash. Is None by default.
+        hash (Optional[str]): Hash of the file. Is None by default.
+
+    Returns:
+        documents (list[MetadataEntity]): List of all documents that match
+            parameters in the database
+
+    """
+    documents = []
+    search = {}
+
+    try:
+        if file_name:
+            search["name"] = file_name
+        if hash and hash_type:
+            hash_parameter = "hashes." + str(hash_type)
+            search[hash_parameter] = hash
+        elif hash_type:
+            raise HTTPException(status_code=400, detail="Enter hash")
+        elif hash:
+            raise HTTPException(status_code=400, detail="Enter hash type")
+        async for doc in collection.find(search):
+            doc["id"] = str(doc["_id"])
+            documents.append(MetadataEntity(**doc))
+        return documents
+
+    except errors.ServerSelectionTimeoutError:
+        raise HTTPException(
+            status_code=408,
+            detail=(
+                "The computer can't find the server.",
+                "Check the IP Address and the server name.",
+            ),
+        )
+
+
+@router.get("/{id}")
+async def get_metadata_by_id(
+    id: str,
```

Comment:
If this is a mongodb objectId, you should label it as such so it can be validated both client and server side.